### PR TITLE
Fixes #24812 - do not delete managed hosts

### PR DESCRIPTION
**Pull request by jlsherrill**
when running clean_backend_objects

### CODE REVIEW

**Comment by beav** [comment]
@jlsherrill code looks good, will test now.

This is a good change to get in, I'm surprised it hasn't come up as an issue sooner.

File: lib/katello/tasks/clean_backend_objects.rake [hunk 1]
```diff
@@ -72,6 +72,14 @@ namespace :katello do
       end
     end
 
+    def host_unregister_options(host)
+      if host.managed? || host.compute_resource
```
Comment (beav) [inline lib/katello/tasks/clean_backend_objects.rake]: I would log "skipping Foreman record deletion for host <host.id>; host is managed by Foreman" or something similar.

**Comment by beav** [comment]
@jlsherrill one comment about logging, tests pass and code looks good.

**Comment by beav** [comment]
test fails are

```

Error:
Katello::CleanBackendObjectsTest#test_compute_resource_host:
StandardError: No fixture named 'bigcompute' found for fixture set 'compute_resources'
    /var/lib/workspace/workspace/katello-pr-test/test/lib/tasks/clean_backend_objects_test.rb:53:in `test_compute_resource_host'

bin/rails test /var/lib/workspace/workspace/katello-pr-test/test/lib/tasks/clean_backend_objects_test.rb:50

Katello::CleanBackendObjectsTest#test_orphaned_no_commit = 0.61 s = .
Katello::CleanBackendObjectsTest#test_missing_pulp_consumer_commit = 0.85 s = .
Katello::CleanBackendObjectsTest#test_managed_host = 0.48 s = F


Failure:
Katello::CleanBackendObjectsTest#test_managed_host [/var/lib/workspace/workspace/katello-pr-test/lib/katello/tasks/clean_backend_objects.rake:45]:
unexpected invocation: Katello::RegistrationManager.unregister_host(#<Host::Managed:0x1152cf40>, {})
unsatisfied expectations:
- expected exactly twice, not yet invoked: Katello::RegistrationManager.unregister_host(#<Host::Managed:0xa060e50>, {:unregistering => true})
satisfied expectations:
- expected exactly once, invoked once: Katello::Ping.ping(any_parameters)
- allowed any number of times, not yet invoked: Katello::Ping.ping(any_parameters)
- allowed any number of times, not yet invoked: Cert::Certs.ssl_client_key(any_parameters)
- allowed any number of times, not yet invoked: Cert::Certs.ssl_client_cert(any_parameters)
- allowed any number of times, not yet invoked: Cert::Certs.ca_cert(any_parameters)
- expected exactly once, invoked once: Katello::Resources::Candlepin::Consumer.all_uuids(any_parameters)
- expected exactly once, invoked once: #<Mock:0x11435c18>.retrieve_all(any_parameters)
- allowed any number of times, invoked once: #<Mock:0x11434908>.consumer(any_parameters)
- allowed any number of times, invoked once: #<Mock:0x1143f830>.extensions(any_parameters)
- allowed any number of times, invoked once: Katello.pulp_server(any_parameters)
```

**Comment by jlsherrill** [comment]
@beav updated!

File: lib/katello/tasks/clean_backend_objects.rake [hunk 2]
```diff
@@ -72,6 +72,15 @@ namespace :katello do
...
+        print "Leaving #{host.name} record in place, it is either managed or assigned to a compute resource."
```
Comment (beav) [inline lib/katello/tasks/clean_backend_objects.rake]: I would say "Foreman record" just so people know the Pulp + Katello + Candlepin records are expected to disappear

**Comment by beav** [comment]
@jlsherrill small comment but APJ otherwise

**Pull request merged**